pennylane/capture/switches.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the switches to (de)activate the capturing mechanism, and a
+status reporting function on whether it is enabled or not.
+"""
+from typing import Callable
+
+has_jax = True
+try:
+    import jax  # pylint: disable=unused-import
+except ImportError:
+    has_jax = False
+
+
+def _make_switches() -> [Callable[[], None], Callable[[], None], Callable[[], bool]]:
```

Comment:
Very cool implementation! Never thought about this, but this effectively hides the "global" flag from manual mutation. 😍